### ISSUE TITLE
Set up bot like Servo's @highfive to @-mention people when files in a subdirectory are modified?

### ISSUE THREAD

**Issue by fitzgen** [bytecodealliance/wasmtime#1234]
The @highfive bot lets people opt into getting @-mentioned whenever a pull request modifies files in some configured subdirectory. This would allow, for example, folks only interested in Cranelift stuff to not follow the whole repo and all of its issue emails, but still get notified of changes that affect Cranelift.

**Comment by bjorn3** [comment]
It would also be nice to get a notification for issues labeld `cranelift`. Will the bot @-mention be opt-in for everyone, including people who haven't every contributed to Cranelift in any way?

**Comment by fitzgen** [comment]
> Will the bot @-mention be opt-in for everyone, including people who haven't every contributed to Cranelift in any way?

It is on a user-by-user basis. For servo, you send a PR to a repo that holds all the configuration for the bot and add yourself and the subdirectories you're interested in.

**Comment by tschneidereit** [comment]
Relatedly, we can also set up a GitHub Action to auto-label pull requests based on changed files: https://github.com/actions/labeler/blob/master/README.md

**Comment by kubkon** [comment]
However we decide to tackle this, it's a much welcomed change given the increased scope of the repo!

**Comment by fitzgen** [comment]
I created a user for our highfive bot: @bytecodealliance-highfive

and I forked the rust-lang/highfive repo: https://github.com/bytecodealliance/highfive

The next steps would be to

1. Replace the rust configs with our own configs.
1. Add @bytecodealliance-highfive as an admin on relevant repos (mostly this one).
1. Start running the bot on some server somewhere and add a webhook for this repo that points at it.

I can't do those last two steps myself.
